pkg/catalog/catalog.go

```diff
@@ -291,6 +293,32 @@ func ListServices(client *occlient.Client) (ServiceTypeList, error) {
 	}, nil
 }
 
+// ListOperatorServices fetches a list of Operators from the cluster and
+// returns only those Operators which are successfully installed on the cluster
+func ListOperatorServices() (*olm.ClusterServiceVersionList, error) {
+	client, err := kclient.New()
```

Comment:
We can pass this client as a parameter

Reply:
As a parameter to what? Not sure if I understand you.

Reply:
Yes like 
```
ListOperatorServices(client kclient.Client)
```
and using it like 
```
catalog.ListOperatorServices(*o.KClient)
```

Reply:
>      catalog.ListOperatorServices(*o.KClient)

What I understood from the discussions we had around refactoring was that we intend to avoid passing around the client in the code and initiate one when needed.

Reply:
> > ```
> >  catalog.ListOperatorServices(*o.KClient)
> > ```
> 
> What I understood from the discussions we had around refactoring was that we intend to avoid passing around the client in the code and initiate one when needed.

That would require a singleton pattern and currently we don't have that for the clients.

Reply:
Yes we don't have that currently. But through this PR, I'm trying to move towards that pattern in small steps instead of refactoring the entire `service`/`catalog` package in one shot.

Reply:
But I remember people having concerns with the singleton pattern. If the architecture design changes to a initialize before mechanism, we might need to change this again as the client has already been initialized in the CLI layer before.

Reply:
I agree with mrinal here, lets pass the client for now we are not sure about it yet.

Reply:
Ah okay. For some reason, I had the impression that we liked the singleton idea for client. I'll change this. Thanks @mik-dass & @adisky ! :+1:

Reply:
Done.